### ISSUE TITLE
[Bug] Multitask training encount 0 loss value

### ISSUE THREAD

**Issue by Purewhiter** [open-mmlab/mmpretrain#1688]
### Branch

main branch (mmpretrain version)

### Describe the bug

I use multitaskDataset,but while training, the losses printed are always 0，for all subtasks. I debug step by step and found that the MultiTaskDataSample in dataloader has nothing, e.g. metainfo or tasks is a empty dict {}.

Here is the running output:
07/05 18:53:26 - mmengine - INFO - Epoch(train)   [1][10/10]  lr: 1.0000e-01  eta: 0:20:43  time: 0.6248  data_time: 0.3374  memory: 1876  loss: 0.0000  toe_loss: 0.0000  toe_mask_size: 0.0000  vamp_loss: 0.0000  vamp_mask_size: 0.0000
07/05 18:53:26 - mmengine - INFO - Saving checkpoint at 1 epochs
07/05 18:53:32 - mmengine - INFO - Epoch(val) [1][10/10]    toe_Accuracy: 0  vamp_Accuracy: 0  data_time: 0.3198  time: 0.4791

the following is my config file:

# model settings
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='ResNet_CIFAR',
        depth=50,
        num_stages=4,
        out_indices=(3, ),
        style='pytorch'),
    neck=dict(type='GlobalAveragePooling'),
    head=dict(
        type='MultiTaskHead',
        task_heads=dict(
            toe=dict(
                type='LinearClsHead',
                num_classes=10,
                in_channels=2048,
                loss=dict(type='CrossEntropyLoss', loss_weight=1.0)
            ),
            vamp=dict(
                type='LinearClsHead',
                num_classes=10,
                in_channels=2048,
                loss=dict(type='CrossEntropyLoss', loss_weight=1.0)
            )
    )))

# dataset settings
dataset_type = 'MultiTaskDataset'
data_preprocessor = dict(
    num_classes=10,
    # RGB format normalization parameters
    mean=[125.307, 122.961, 113.8575],
    std=[51.5865, 50.847, 51.255],
    # loaded images are already RGB format
    to_rgb=False)

train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='RandomCrop', crop_size=256, padding=4),
    dict(type='RandomFlip', prob=0.5, direction='horizontal'),
    dict(type='PackMultiTaskInputs',
         multi_task_fields=['toe', 'vamp'],
         ),
]

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='PackMultiTaskInputs',
         multi_task_fields=['toe', 'vamp'],
         ),
]

train_dataloader = dict(
    batch_size=1,
    num_workers=1,
    dataset=dict(
             type="MultiTaskDataset",
             ann_file="annotation.json",
             data_root="padding_result3",
             # the `data_prefix` is not required since all paths are
             # relative to the `data_root`.
             pipeline=train_pipeline,
         ),
    sampler=dict(type='DefaultSampler', shuffle=True),
)

val_dataloader = dict(
    batch_size=1,
    num_workers=1,
    dataset=dict(
             type="MultiTaskDataset",
             ann_file="annotation.json",
             data_root="padding_result3",
             # the `data_prefix` is not required since all paths are
             # relative to the `data_root`.
             pipeline=test_pipeline,
         ),
    sampler=dict(type='DefaultSampler', shuffle=False),
)
val_evaluator = dict(type='MultiTasksMetric',
                     task_metrics=dict(
                         toe=[dict(type='Accuracy', topk=(1, ))],
                         vamp=[dict(type='Accuracy', topk=(1, ))],
                     )
                    )

test_dataloader = val_dataloader
test_evaluator = val_evaluator

### Environment

{'sys.platform': 'win32',
 'Python': '3.10.11 | packaged by Anaconda, Inc. | (main, May 16 2023, '
           '00:55:32) [MSC v.1916 64 bit (AMD64)]',
 'CUDA available': True,
 'numpy_random_seed': 2147483648,
 'GPU 0': 'NVIDIA GeForce RTX 4060 Laptop GPU',
 'CUDA_HOME': 'E:\\NVIDIA GPU Computing Toolkit\\CUDA\\v11.8',
 'NVCC': 'Cuda compilation tools, release 11.8, V11.8.89',
 'MSVC': '用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.35.32217.1 版',
 'GCC': 'n/a',
 'PyTorch': '2.0.1',
 'TorchVision': '0.15.2',
 'OpenCV': '4.7.0',
 'MMEngine': '0.7.4',
 'MMCV': '2.0.0',
 'MMPreTrain': '1.0.0rc8+4dd8a86'}

**Comment by Ezra-Yu** [comment]
Generally, the loss can not be 0, This may be caused by the head not reading the gt_label

**Comment by jarsvn** [comment]
@Purewhiter 

I also encountered the same error as you. Have you fixed it yet? Please share it with me.

**Comment by jarsvn** [comment]
I've fixed it. The configuration was wrong in the 'PackMultiTaskInputs' section. Here is my configuration file.

_base_ = [
    '../_base_/models/resnet50.py', '../_base_/datasets/imagenet_bs32.py',
    '../_base_/schedules/imagenet_bs256.py', '../_base_/default_runtime.py'
]

# runtime settings
log_level = 'DEBUG'

# model settings
model = dict(
    backbone=dict(
        frozen_stages=2,
        init_cfg=dict(
            type='Pretrained',
            checkpoint='https://download.openmmlab.com/mmclassification/v0/resnet/resnet50_8xb32_in1k_20210831-ea4938fc.pth',
            prefix='backbone',
        )),
    head=dict(
        type='MultiTaskHead',
        task_heads=dict(
            tilt=dict(
                in_channels=2048, 
                num_classes=9, 
                loss=dict(type='CrossEntropyLoss', loss_weight=1.0), 
                topk=(1, 5,),
                type='LinearClsHead'),
            pan=dict(
                in_channels=2048, 
                num_classes=13, 
                loss=dict(type='CrossEntropyLoss', loss_weight=1.0), 
                topk=(1, 5,),
                type='LinearClsHead')),
        _delete_=True),
)

# data settings
dataset_type = 'MultiTaskDataset'

data_preprocessor = dict(
    to_rgb=False)

train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(scale=224, type='RandomResizedCrop'),
    dict(type='PackMultiTaskInputs', multi_task_fields=('gt_label', )),
]

train_dataloader = dict(
    batch_size=16,
    num_workers=1,
    dataset=dict(
        ann_file='annotation/train.json',
        data_root='data/head_pose',
        data_prefix='train',
        pipeline=train_pipeline,
        type=dataset_type),
    sampler=dict(shuffle=False, type='DefaultSampler'),
    _delete_=True)
    
val_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(edge='short', scale=256, type='ResizeEdge'),
    dict(crop_size=224, type='CenterCrop'),
    dict(type='PackMultiTaskInputs', multi_task_fields=('gt_label', )),
]

val_dataloader = dict(
    batch_size=16,
    num_workers=1,
    dataset=dict(
        ann_file='annotation/test.json',
        data_root='data/head_pose',
        data_prefix='test',
        pipeline=val_pipeline,
        type=dataset_type),
    sampler=dict(shuffle=False, type='DefaultSampler'),
    _delete_=True)
        
val_evaluator = dict(
    type='MultiTasksMetric',
    task_metrics=dict(
        tilt=[dict(type='Accuracy', topk=(1, 5,))],
        pan=[dict(type='Accuracy', topk=(1, 5,))],
    ),
    _delete_=True
)        
        
test_dataloader = val_dataloader
test_evaluator = val_evaluator

**Comment by ccxi1008** [comment]
> I've fixed it. The configuration was wrong in the 'PackMultiTaskInputs' section. Here is my configuration file.
> 
> _base_ = [ '../_base_/models/resnet50.py', '../_base_/datasets/imagenet_bs32.py', '../_base_/schedules/imagenet_bs256.py', '../_base_/default_runtime.py' ]
> 
> # runtime settings
> log_level = 'DEBUG'
> 
> # model settings
> model = dict( backbone=dict( frozen_stages=2, init_cfg=dict( type='Pretrained', checkpoint='https://download.openmmlab.com/mmclassification/v0/resnet/resnet50_8xb32_in1k_20210831-ea4938fc.pth', prefix='backbone', )), head=dict( type='MultiTaskHead', task_heads=dict( tilt=dict( in_channels=2048, num_classes=9, loss=dict(type='CrossEntropyLoss', loss_weight=1.0), topk=(1, 5,), type='LinearClsHead'), pan=dict( in_channels=2048, num_classes=13, loss=dict(type='CrossEntropyLoss', loss_weight=1.0), topk=(1, 5,), type='LinearClsHead')), _delete_=True), )
> 
> # data settings
> dataset_type = 'MultiTaskDataset'
> 
> data_preprocessor = dict( to_rgb=False)
> 
> train_pipeline = [ dict(type='LoadImageFromFile'), dict(scale=224, type='RandomResizedCrop'), dict(type='PackMultiTaskInputs', multi_task_fields=('gt_label', )), ]
> 
> train_dataloader = dict( batch_size=16, num_workers=1, dataset=dict( ann_file='annotation/train.json', data_root='data/head_pose', data_prefix='train', pipeline=train_pipeline, type=dataset_type), sampler=dict(shuffle=False, type='DefaultSampler'), _delete_=True)
> 
> val_pipeline = [ dict(type='LoadImageFromFile'), dict(edge='short', scale=256, type='ResizeEdge'), dict(crop_size=224, type='CenterCrop'), dict(type='PackMultiTaskInputs', multi_task_fields=('gt_label', )), ]
> 
> val_dataloader = dict( batch_size=16, num_workers=1, dataset=dict( ann_file='annotation/test.json', data_root='data/head_pose', data_prefix='test', pipeline=val_pipeline, type=dataset_type), sampler=dict(shuffle=False, type='DefaultSampler'), _delete_=True)
> 
> val_evaluator = dict( type='MultiTasksMetric', task_metrics=dict( tilt=[dict(type='Accuracy', topk=(1, 5,))], pan=[dict(type='Accuracy', topk=(1, 5,))], ), _delete_=True )
> 
> test_dataloader = val_dataloader test_evaluator = val_evaluator

Same issue.How you solve it?

**Comment by ccxi1008** [comment]
> ### Branch
> main branch (mmpretrain version)
> 
> ### Describe the bug
> I use multitaskDataset,but while training, the losses printed are always 0，for all subtasks. I debug step by step and found that the MultiTaskDataSample in dataloader has nothing, e.g. metainfo or tasks is a empty dict {}.
> 
> Here is the running output: 07/05 18:53:26 - mmengine - INFO - Epoch(train) [1][10/10] lr: 1.0000e-01 eta: 0:20:43 time: 0.6248 data_time: 0.3374 memory: 1876 loss: 0.0000 toe_loss: 0.0000 toe_mask_size: 0.0000 vamp_loss: 0.0000 vamp_mask_size: 0.0000 07/05 18:53:26 - mmengine - INFO - Saving checkpoint at 1 epochs 07/05 18:53:32 - mmengine - INFO - Epoch(val) [1][10/10] toe_Accuracy: 0 vamp_Accuracy: 0 data_time: 0.3198 time: 0.4791
> 
> the following is my config file:
> 
> # model settings
> model = dict( type='ImageClassifier', backbone=dict( type='ResNet_CIFAR', depth=50, num_stages=4, out_indices=(3, ), style='pytorch'), neck=dict(type='GlobalAveragePooling'), head=dict( type='MultiTaskHead', task_heads=dict( toe=dict( type='LinearClsHead', num_classes=10, in_channels=2048, loss=dict(type='CrossEntropyLoss', loss_weight=1.0) ), vamp=dict( type='LinearClsHead', num_classes=10, in_channels=2048, loss=dict(type='CrossEntropyLoss', loss_weight=1.0) ) )))
> 
> # dataset settings
> dataset_type = 'MultiTaskDataset' data_preprocessor = dict( num_classes=10, # RGB format normalization parameters mean=[125.307, 122.961, 113.8575], std=[51.5865, 50.847, 51.255], # loaded images are already RGB format to_rgb=False)
> 
> train_pipeline = [ dict(type='LoadImageFromFile'), dict(type='RandomCrop', crop_size=256, padding=4), dict(type='RandomFlip', prob=0.5, direction='horizontal'), dict(type='PackMultiTaskInputs', multi_task_fields=['toe', 'vamp'], ), ]
> 
> test_pipeline = [ dict(type='LoadImageFromFile'), dict(type='PackMultiTaskInputs', multi_task_fields=['toe', 'vamp'], ), ]
> 
> train_dataloader = dict( batch_size=1, num_workers=1, dataset=dict( type="MultiTaskDataset", ann_file="annotation.json", data_root="padding_result3", # the `data_prefix` is not required since all paths are # relative to the `data_root`. pipeline=train_pipeline, ), sampler=dict(type='DefaultSampler', shuffle=True), )
> 
> val_dataloader = dict( batch_size=1, num_workers=1, dataset=dict( type="MultiTaskDataset", ann_file="annotation.json", data_root="padding_result3", # the `data_prefix` is not required since all paths are # relative to the `data_root`. pipeline=test_pipeline, ), sampler=dict(type='DefaultSampler', shuffle=False), ) val_evaluator = dict(type='MultiTasksMetric', task_metrics=dict( toe=[dict(type='Accuracy', topk=(1, ))], vamp=[dict(type='Accuracy', topk=(1, ))], ) )
> 
> test_dataloader = val_dataloader test_evaluator = val_evaluator
> 
> ### Environment
> {'sys.platform': 'win32', 'Python': '3.10.11 | packaged by Anaconda, Inc. | (main, May 16 2023, ' '00:55:32) [MSC v.1916 64 bit (AMD64)]', 'CUDA available': True, 'numpy_random_seed': 2147483648, 'GPU 0': 'NVIDIA GeForce RTX 4060 Laptop GPU', 'CUDA_HOME': 'E:\NVIDIA GPU Computing Toolkit\CUDA\v11.8', 'NVCC': 'Cuda compilation tools, release 11.8, V11.8.89', 'MSVC': '用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.35.32217.1 版', 'GCC': 'n/a', 'PyTorch': '2.0.1', 'TorchVision': '0.15.2', 'OpenCV': '4.7.0', 'MMEngine': '0.7.4', 'MMCV': '2.0.0', 'MMPreTrain': '1.0.0rc8+4dd8a86'}
> 
> ### Other information
> _No response_

Hi！ Do you know how to test the model and visualize  the results?
